src/python_testing/TC_DRLK_2_9.py

```diff
@@ -0,0 +1,793 @@
+            try:
+                await self.send_single_cmd(cmd=drlkcluster.Commands.SetUser(
+                    operationType=Clusters.DoorLock.Enums.DataOperationTypeEnum.kAdd,
+                    userIndex=userIndex_2,
+                    userName=user_name,
+                    userUniqueID=user_unique_id,
+                    userStatus=Clusters.DoorLock.Enums.UserStatusEnum.kOccupiedEnabled,
+                    userType=Clusters.DoorLock.Enums.UserTypeEnum.kUnrestrictedUser,
+                    credentialRule=Clusters.DoorLock.Enums.CredentialRuleEnum.kSingle),
+                    endpoint=self.app_cluster_endpoint,
+                    timedRequestTimeoutMs=1000)
+            except InteractionModelError as e:
+                logging.exception(e)
+
+        self.step("11")
+        if self.pics_guard(self.check_pics("DRLK.S.F00") and self.check_pics("DRLK.S.F08")
+                           and self.check_pics("DRLK.S.C22.Rsp") and self.check_pics("DRLK.S.C23.Tx")):
+            await self.set_credential_cmd(credentialData=self.pin_code,
+                                          operationType=drlkcluster.Enums.DataOperationTypeEnum.kAdd,
+                                          credential_enum=drlkcluster.Enums.CredentialTypeEnum.kPin,
+                                          credentialIndex=credentialIndex_1, userIndex=userIndex_2, userStatus=NullValue, userType=NullValue, statuscode=Status.Success)
+        self.step("12a")
+        if self.pics_guard(self.check_pics("DRLK.S.F00") and self.check_pics("DRLK.S.F08") and self.check_pics("DRLK.S.C26.Rsp")):
+            credentials = drlkcluster.Structs.CredentialStruct(credentialIndex=0xFFFE,
+                                                               credentialType=drlkcluster.Enums.CredentialTypeEnum.kPin)
+            await self.clear_credentials_cmd(credential=credentials)
+
+        self.step("13")
+        if self.pics_guard(self.check_pics("DRLK.S.F00") and self.check_pics("DRLK.S.F08") and self.check_pics("DRLK.S.C24.Rsp") and self.check_pics("DRLK.S.C25.Tx")):
+            await self.get_credentials_status(credentialIndex=credentialIndex_1,
+                                              credentialType=drlkcluster.Enums.CredentialTypeEnum.kPin, credential_exists=False, next_credential_index=NullValue)
+        self.step("14a")
+        if self.pics_guard(self.check_pics("DRLK.S.F08") and self.check_pics("DRLK.S.C26.Rsp")):
+            feature_map = await self.read_attributes_from_dut(endpoint=self.app_cluster_endpoint,
+                                                              cluster=drlkcluster,
+                                                              attribute=Clusters.DoorLock.Attributes.FeatureMap)
+            aliro_enabled = feature_map & Clusters.DoorLock.Bitmaps.Feature.kAliroProvisioning
+        if (aliro_enabled):
+            credentials = drlkcluster.Structs.CredentialStruct(credentialIndex=credentialIndex_1,
```

Comment:
Assignment of 'credentials' needs to be moved before the 'if (aliro_enabled)' line. Otherwise, the 'clearCredential' command in the else clause uses credentials list from step 12

Reply:
corrected